src/main/java/com/taskadapter/redmineapi/bean/Project.java

```diff
@@ -99,13 +98,29 @@ public void setName(String name) {
      * @return Trackers allowed in this project (e.g.: Bug, Feature, Support, Task, ...)
      */
     public Collection<Tracker> getTrackers() {
+    	if (!storage.isPropertySet(TRACKERS)) //checks because trackers storage is not created for new projects
+    		return new HashSet<>();
         return Collections.unmodifiableCollection(storage.get(TRACKERS));
     }
 
+    /**
+     * Adds the specified trackers to this project.
+     * If this project is created or updated on the redmine server, 
+     * each tracker id must be a valid tracker on the server.
+     */
     public void addTrackers(Collection<Tracker> trackers) {
+    	if (!storage.isPropertySet(TRACKERS)) //checks because trackers storage is not created for new projects
+    		this.clearTrackers();
```

Comment:
if property is not set, then clear something - this is confusing. 
why clear if it is not set?

correspondingly, similar comment about the change in getTrackers() method. 
can we keep existing ```storage.set(TRACKERS, new HashSet<>());``` (I see it was removed here)   
and use that one instead?

Reply:
what about this change?

Reply:
Sorry, I forgot commit the comment.
addTrackers(): I have replaced this.clearTrackers() by storage.set(TRACKERS, new HashSet<>())

getTrackers(): new HashSet<>() is returned in order to avoid modifying the bean (as it is a getter). If we set the storage to an empty set of trackers and next operation is creating the project, it will be created without default trackers.

Reply:
I see the idea, but I am worried about getTrackers() method returning fake data. 
users can attempt to call
.getTrackers() and then add an item to the collection, and the behavior would depend on whether or not there were trackers before: it would return either a new **modifiable** hashset, or an **unmodifiable** set if there were some trackers stored already.
one option would be to return an unmodifiable empty set

Reply:
I agree. Then, the change should be
```
    public Collection<Tracker> getTrackers() {
    	if (!storage.isPropertySet(TRACKERS)) //checks because trackers storage is not created for new projects
    		return Collections.unmodifiableCollection(new HashSet<Tracker>());
        return Collections.unmodifiableCollection(storage.get(TRACKERS));
    }
```
Is this right?

Reply:
this would work

Reply:
Done. Thank you for your review.